libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.ts

```diff
@@ -163,36 +203,95 @@ export class ChartJSService {
   }
 
   private createOptionsObject(args: {
-    type: ChartType;
     customOptions?: ChartOptions;
     annotations?: AnnotationOptions[];
+    chartPeriod?: ChartPeriod;
+    dataLabelOptions?: ChartDataLabelOptions;
   }): ChartOptions {
-    const { type, customOptions, annotations } = args;
+    const {
+      customOptions,
+      annotations,
+      chartPeriod,
+      dataLabelOptions: chartDataLabelOptions,
+    } = args;
 
-    const typeConfig = this.chartConfigService.getTypeConfig(type);
+    const typeConfig = this.chartConfigService.getTypeConfig(this.chartType);
     const typeConfigOptions = typeConfig?.options;
     const annotationPluginOptions = annotations
       ? this.createAnnotationPluginOptionsObject(annotations)
       : {};
+
+    let defaultOptions: ChartOptions = {
+      locale: this.locale,
+      plugins: { tooltip: { callbacks: {} } },
+      scales: { y: { ticks: {} } },
+    };
+
+    if (this.chartType === 'stock') {
+      defaultOptions.plugins.tooltip.callbacks.label = (context) => {
+        // It's not possible to add spacing between color legend and text so we
+        // prefix with a space.
+        return ' ' + context.formattedValue + (chartDataLabelOptions.valueSuffix || '');
+      };
+
+      defaultOptions.scales.y.ticks.callback = (value) => {
+        return value + (chartDataLabelOptions.valueSuffix || '');
+      };
+
+      defaultOptions.plugins.tooltip.callbacks.title = (tooltipItems) => {
+        const date = toDate((tooltipItems[0]?.raw as any)?.x);
+        if (date.valueOf()) {
+          return date.toLocaleTimeString(this.locale, {
+            day: 'numeric',
+            month: 'short',
+            hour: '2-digit',
+            minute: '2-digit',
+          });
+        }
+      };
+    }
+
     let options: ChartOptions = mergeDeepAll(
+      defaultOptions,
       typeConfigOptions,
       customOptions,
       annotationPluginOptions
     );
+
     return this.applyInteractionFunctionsExtensions(options);
   }
 
+  private getDefaultStockLabels(datasets, locale: ChartLocale) {
+    const largestDataset = datasets.reduce((previousDataset, currentDataset) =>
+      previousDataset.data.length > currentDataset.data.length ? previousDataset : currentDataset
+    );
+    return largestDataset.data.map((point) =>
+      toDate(point.x).toLocaleDateString(locale, {
+        month: 'short',
+        day: 'numeric',
+      })
+    );
+  }
+
```

Comment:
Consider typing the function 😄 
`point['x']` should fix the type error caused by setting `datasets: ChartDataset[]`.
```suggestion
  private getDefaultStockLabels(datasets: ChartDataset[], locale: ChartLocale) {
    const largestDataset = datasets.reduce((previousDataset, currentDataset) =>
      previousDataset.data.length > currentDataset.data.length ? previousDataset : currentDataset
    );
    return largestDataset.data.map((point) =>
      toDate(poin['x']).toLocaleDateString(locale, {
        month: 'short',
        day: 'numeric',
      })
    );
  }

```

Reply:
point: ScatterPoint also did the trick.